### DISCUSSION TITLE
C4GT: Digital Learner's Passbook

### DISCUSSION THREAD

**Discussion by cchaitanya27** [Sunbird-Ed/Community#487] [Feature request]
### Project Details
Sunbird ED is a platform that can be used to deploy scale-ready solutions in the area of learning, capacity building, and professional development. The platform provides capabilities for a user to register and take trainings that are hosted on the platform. The trainings can also have certificates issued. Each learner has a profile that lists all the courses taken by him/her.

This is to extend this capability to create a pluggable Digital Learner Passbook. A digital learner passbook will act as a digital profile & docket of a learner wherein it will comprise details of the learner along with all the digitally verifiable credentials and will also capture the skills/competencies that have been achieved by the learner. The verifiable credentials can be issued by different sources, not just the Sunbird ED solution.
Refer to [this document](https://docs.google.com/document/d/1zuek44e-L9aI5jAxvoODAgSzlvXY9j_Z/edit?usp=sharing&ouid=102564518876064968464&rtpof=true&sd=true) for the details.
### Features to be implemented
A learner passbook service implemented using Sunbird RC that contains following details
Basic demographic details of the learner 
List of Verifiable Credentials mapped to competencies or learning outcomes
The Verifiable Credentials come from [DigiLocker](https://directory.apisetu.gov.in/directory/api/digilocker)
The Verifiable Credentials issued Sunbird Ed platform are integrated with DigiLocker

### What exists
Sunbird RC that can be used to create registry
Sunbird ED platform that has
User Registration and Profile
Verifiable Certificates issued after a Course completion

### What needs to be built
1. Developing an interface which will:
a. Leverage DigiLocker APIs to extract learner progress related information – with user consent via OTP e.g. 10th Board marksheet, 12th Board marksheet etc.
b. Process the information and generate 1-pager summarised view (Learner Progress; Benefits Passbook – LPBP) of the achievements & competency level of the learner
c. Make the LPBP shareable via digital mediums
2. The assessor, through the LPBP, shall be able to verify the relevant credentials via DigiLocker – verify the claims made by the learner ; with user consent via OTP

### Learning Path
Complexity: Medium
Skills Required: 
Understanding existing Sunbird ED, Sunbird Learn, Sunbird RC
Java, Node, Angular
Mentors: Niharika, Chinmay
Project size: 6 weeks

### Milestones
Detailing and documenting the capabilities
Creating the design of the Digital Learner's Passbook service
Implementation of the Digital Learner's Passbook service
Unit Test cases

**Comment by chitranshu-keshav** [comment]
@Sunbird-Ed/sunbirded_workinggroup  Can you please review and provide your inputs on this feature request

**Comment by JeevaDev7** [comment]
HI @chitranshu-keshav  @anandp504  I am interested in this project how can I Contribute ?

**Reply by vinukumar-vs** [reply]
@cchaitanya27 We got email from **Nikita**(@nikitamahoviya) showing interest in this problem statement.

**Reply by kameshbhr** [reply]
@JeevaRP-SDE, @nikitamahoviya Please check out the C4GT website on how you can contribute: https://www.codeforgovtech.in/. You can join the Discord channel provided there to raise your queries and for further discussions.

**Comment by Mounika-Pavanthi** [comment]
I am thrilled to contribute to the Digital Learner Passbook project and be part of the team that enhances the capabilities of Sunbird ED. With my expertise in Java, Node, and Angular, combined with a deep understanding of Sunbird platforms, I am confident in my ability to develop the Digital Learner Passbook service. I am dedicated to delivering high-quality code, collaborating with mentors, and achieving project milestones within the allocated timeframe. Please consider my involvement in the project, as I am passionate about empowering learners through digital profiles and verifiable credentials.

**Comment by Shyamalsian** [comment]
Hiii there, If no one is assigned to the work can I contribute in the project.
